src/radical/pilot/pmgr/launching/default.py

```diff
@@ -1119,6 +1121,23 @@ def _prepare_pilot(self, resource, rcfg, pilot, expand):
             if virtenv_mode != 'private':
                 cleanup = cleanup.replace('v', '')
 
+        # don't stage sdists on 'installed', 'local'
+        if virtenv_mode in ['installed', 'local']: stage_sdists = False
+        else                                     : stage_sdists = True
+
+        # use local VE ?
+        if virtenv_mode == 'local':
```

Comment:
we can reassign it to `'use'` value:
```python
if virtenv_mode == 'local':
    virtenv_mode = 'use'
    ... 
    # check env var, set virtenv and python_dist
```
and then we don't need new value for `virtenv_mode` in `bootstrap_0.sh`

Reply:
Yeah, I had that at some point, but worried that it is too obfuscated when reading  / debugging the bootstrapper, and thus rather would leave it explicit.

Reply:
don't mind both, but I would treat `'local'` value as a flag parameter to set all others including itself (`python_dist`, `virtenv_mode`, `virtenv`, `rp_version`)